### ISSUE TITLE
When will the use_distributed_optimizer mode, namely zero-1, in Megatron-LM be supported?

### ISSUE THREAD

**Issue by LeoAtlanto** [intelligent-machine-learning/dlrover#949]
Firstly, thanks for the amazing project!

I'm recently working on a project using [Megatron-LM](https://github.com/NVIDIA/Megatron-LM) with [`args.use_distributed_optimizer`](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/checkpointing.py#L258) on, or in other words, [ZeRO](https://arxiv.org/abs/1910.02054)-1 mode, where the overhead of [`optimizer.save_parameter_state(optim_checkpoint_name)`](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/checkpointing.py#L262) which turned out to be the main part could not be saved, and I'm wondering will or when it could be possible, thanks!

**Comment by workingloong** [comment]
Megatron-LM rank 0 gathers all optimizer tensors from other ranks using gloo as the communication backend. It is much slower than NCCL. You could set the communication group with NCCL to gather all optimizer tensors wich may be mush faster than gloo.

[data_parallel_group_gloo = mpu.get_data_parallel_group_gloo(with_context_parallel=True)](https://github.com/NVIDIA/Megatron-LM/blob/main/megatron/optimizer/distrib_optimizer.py#L695C9-L695)

Nevertheless, the optimal solution is that each rank saves its optimizer shard which needs some work.  DeepSpeed have implemented it in its ZERO, and FSDP also implemented the features.

**Comment by workingloong** [comment]
We have released dlrover[torch]==0.3.4 to save and load the distributed optimizer in parallel. You can `pip install dlrover[torch]==0.3.4 -U`  and use the APIs in `megatron/training.py`.

```python
# from megatron.checkpointing import load_checkpoint
# from megatron.checkpointing import save_checkpoint

from dlrover.trainer.torch.flash_checkpoint.megatron_dist_ckpt import save_checkpoint
from dlrover.trainer.torch.flash_checkpoint.megatron_dist_ckpt import load_checkpoint
```
